tests/product/image_builder.py

```diff
@@ -0,0 +1,203 @@
+            return [slave_image_name, master_image_name]
+        elif cluster_type == self.STANDALONE_PRESTO_CLUSTER:
+            slave_image_name = 'teradatalabs/pa_test/nohadoop_presto_slave'
+            master_image_name = 'teradatalabs/pa_test/nohadoop_presto_master'
+            return [slave_image_name, master_image_name]
+        elif cluster_type == self.PA_ONLY_YS_CLUSTER:
+            slave_image_name = 'teradatalabs/pa_test/nohadoop_pa_only_ys_slave'
+            master_image_name = 'teradatalabs/pa_test/nohadoop_pa_only_ys_master'
+            return [slave_image_name, master_image_name]
+        else:
+            return []
+
+    def _assert_expected_images_created(self, cluster_type):
+        docker_images = check_output(['docker', 'images'])
+        expected_images = self._get_expected_image_names(cluster_type)
+        for expected_image in expected_images:
+            assert expected_image in docker_images
+
+    def _setup_cluster_with_no_hadoop_provider_assert_images(self, cluster_type):
+        self._setup_cluster(NoHadoopBareImageProvider(),
+                            cluster_type)
+        self._assert_expected_images_created(cluster_type)
+
+    def test_setup_standalone_presto_images(self):
+        cluster_type = self.STANDALONE_PRESTO_CLUSTER
+        self._setup_cluster_with_no_hadoop_provider_assert_images(cluster_type)
+
+    def test_setup_standalone_pa_only_images(self):
+        cluster_type = self.PA_ONLY_CLUSTER
+        self._setup_cluster_with_no_hadoop_provider_assert_images(cluster_type)
+
+    def test_setup_standalone_bare_images(self):
+        cluster_type = self.BARE_CLUSTER
+        self._setup_cluster_with_no_hadoop_provider_assert_images(cluster_type)
+
+    def test_setup_yarn_slider_pa_only_images(self):
+        cluster_type = self.PA_ONLY_YS_CLUSTER
+        self._setup_cluster_with_no_hadoop_provider_assert_images(cluster_type)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--standalone_presto",
+                        help="Create standalone presto images",
+                        action="store_true")
+    parser.add_argument("--standalone_pa",
+                        help="Create standalone presto-admin only images",
+                        action="store_true")
+    parser.add_argument("--standalone_bare",
+                        help="Create standalone bare images",
+                        action="store_true")
+    parser.add_argument("--yarn_slider_pa",
+                        help="Create yarn-slider presto-admin only images",
+                        action="store_true")
+    parser.add_argument("--all",
+                        help="Create standalone presto, standalone presto-admin only,"
+                             "standalone bare, and yarn-slider presto-admin only images",
+                        action="store_true")
+
+    if len(argv) <= 1:
```

Comment:
I think it is best to re-write the argument parsing to use a positional argument. That way you can more succinctly specify that you require at least one argument and you can specify what the available choices are. Here's what I propose:

```
if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("image_type", metavar="image_type", type=str, nargs="+",
                        choices=["standalone_presto", "standalone_pa", "standalone_bare",
                                 "yarn_slider_pa", "all"], help="Specify the type of "
                        "image to create. The available choices are: standalone_presto, "
                        "standalone_pa, standalone_bare, yarn_slider_pa, all")

    image_builder = ProductTestImageBuilder()

    args = parser.parse_args()
    if "all" in args.image_type:
        image_builder.test_setup_standalone_pa_only_images()
        image_builder.test_setup_standalone_pa_only_images()
        image_builder.test_setup_standalone_bare_images()
        image_builder.test_setup_yarn_slider_pa_only_images()
    else:
        if "standalone_presto" in args.image_type:
            image_builder.test_setup_standalone_presto_images()
        if "standalone_pa" in args.image_type:
            image_builder.test_setup_standalone_pa_only_images()
        if "standalone_bare" in args.image_type:
            image_builder.test_setup_standalone_bare_images()
        if "yarn_slider_pa" in args.image_type:
            image_builder.test_setup_yarn_slider_pa_only_images()
```